alluxio/worker_ring.py

```diff
@@ -170,23 +173,29 @@ def __init__(
         self,
         etcd_hosts=None,
         worker_hosts=None,
+        options=None,
         logger=None,
         num_virtual_nodes=2000,
         max_attempts=100,
         refresh_timeout_seconds=120,
     ):
         self._etcd_hosts = etcd_hosts
-        self._worker_addresses = (
-            WorkerNetAddress.from_worker_hosts(worker_hosts) if worker_hosts else None
-        )
+        self._options = options
+        self._logger = logger or logging.getLogger("ConsistentHashProvider")
         self._num_virtual_nodes = num_virtual_nodes
         self._max_attempts = max_attempts
-        self._logger = logger or logging.getLogger("ConsistentHashProvider")
-
         self._lock = threading.Lock()
         self._is_ring_initialized = False
-        self._update_hash_ring_if_needed(self._worker_addresses)
-        self._start_background_update(refresh_timeout_seconds)
+        self._worker_addresses = None
+        if worker_hosts:
+            self._update_hash_ring_internal(
+                WorkerNetAddress.from_worker_hosts(worker_hosts)
+            )
+        if self._etcd_hosts:
+            self._shutdown_event = threading.Event()
+            self._background_thread = None
+            self._update_hash_ring_etcd()
+            self._start_background_update(refresh_timeout_seconds)
```

Comment:
Per our discussion, could you set it as configurable? Just a simple example: when the timeout given is -1, then we won't refresh the worker list.
BTW: We probably would also need to verify that the timeout given is in the appropriate range.

Reply:
@ChunxuTang the code is still developing, since @jja725 is also working on new code development, i wont do too big range code refactoring